sdks/python/apache_beam/io/hdfs_integration_test/docker-compose.yml

```diff
@@ -0,0 +1,64 @@
+#
+#    Licensed to the Apache Software Foundation (ASF) under one or more
+#    contributor license agreements.  See the NOTICE file distributed with
+#    this work for additional information regarding copyright ownership.
+#    The ASF licenses this file to You under the Apache License, Version 2.0
+#    (the "License"); you may not use this file except in compliance with
+#    the License.  You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+version: '3'
+
+services:
+  # HDFS namenode.
+  namenode:
+    image: uhopper/hadoop-namenode:2.8.1
+    hostname: namenode
+    domainname: hadoop
+    environment:
+      - CLUSTER_NAME=test
+    networks:
+      - test_net
+
+  # HDFS datanode.
+  datanode:
+    image: uhopper/hadoop-datanode:2.8.1
+    hostname: datanode1
+    domainname: hadoop
+    environment:
+      - CLUSTER_NAME=test
+      - CORE_CONF_fs_defaultFS=hdfs://namenode:8020
+    networks:
+      - test_net
+    depends_on:
+      - "namenode"
+
+  # Integration test.
+  test:
+    build:
+      context: ../../../../../
```

Comment:
I would still recommend a smaller context scope to avoid maintaining a complex .dockerignore. For example, sdks/java is in the context although not added to the image.

It is also not obvious that the top-level .dockerignore is for this python test. And if we add similar tests for java or go, we would expect different content to be included.

Reply:
I don't know how to limit the context further. The image requires files under sdks/python and under model/.

Reply:
I removed .dockerignore since it wasn't working with docker-compose anyway.

Reply:
You can copy the content you need into a temporary context directory beforehand. We do this at a smaller scale for the Go code, for example: https://github.com/apache/beam/blob/master/sdks/python/container/pom.xml#L44. Not sure if symlinks would work in this case, if you do something similar in the bash script.

Alternatively, if this is a single machine test, you could map in the beam code as a volume instead and install the SDK/etc at startup instead of as an image build.

Reply:
Symbolic links are not allowed to point outside the build context, so I used copying.